File: app/src/main/java/com/funny/translation/utils/FunnyEachText.java
package com.funny.translation.utils;

import android.content.Context;

import com.funny.translation.FunnyApplication;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.IOException;
import java.nio.charset.StandardCharsets;

public class FunnyEachText {
    static JSONObject WORDS;
    public static void init(Context context) throws IOException, JSONException {
        String assetsData = FileUtil.getAssetsData(context, "words.json");
        WORDS = new JSONObject(new String(assetsData.getBytes(), StandardCharsets.UTF_8));
    }

    public static JSONObject getWords() throws IOException, JSONException {
        if (WORDS == null) {
            init(FunnyApplication.getFunnyContext());
        }
        return WORDS;
    }
}

File: translate/src/main/java/com/funny/translation/translate/utils/FunnyEachText.kt
package com.funny.translation.translate.utils

import android.content.Context
import com.funny.translation.helper.coroutine.Coroutine
import com.funny.translation.helper.readAssets
import com.funny.translation.translate.FunnyApplication
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import org.json.JSONObject
import org.json.JSONException
import java.io.IOException
import java.nio.charset.StandardCharsets

object FunnyEachText {
    var WORDS: JSONObject? = null
    @Throws(IOException::class, JSONException::class)
    suspend fun init(context: Context) {
        withContext(Dispatchers.IO){
            val assetsData: String = context.readAssets("words.json")
            WORDS = JSONObject(String(assetsData.toByteArray(), StandardCharsets.UTF_8))
        }
    }

    @get:Throws(IOException::class, JSONException::class)
    val words: JSONObject
        get() {
            if (WORDS == null) {
                Coroutine.async {
                    init(FunnyApplication.ctx)
                }

            }
            return WORDS!!
        }
}